### ISSUE TITLE
Module not found: Can't resolve 'fs'

### ISSUE THREAD

**Issue by raitucarp** [LyraSearch/plugin-data-persistence#9]
I wanna use this memory persistence feature for my app. But, can't import this plugin into Next.js, always getting below error

```
./node_modules/@lyrasearch/plugin-data-persistence/dist/node/esm/node-bun/index.js:1:0
Module not found: Can't resolve 'fs'
```

**Comment by abhay187** [comment]
Getting the same error on Cloudflare worker besides mentioning it on docs that it works on V8 isolates.

<img width="848" alt="Screenshot 2023-01-05 at 1 05 47 AM" src="https://user-images.githubusercontent.com/61496865/210635385-c7774968-0e0d-41d8-9706-e8320db36f4b.png">

**Comment by mateonunez** [comment]
Hey @raitucarp and @abhay187.

It would be great if we could have a small reproduction of the error that might help people to fix the problem.

Would you like to provide it?

**Comment by abhay187** [comment]
@mateonunez since the issue is with this plugin let's only discuss it here. Here is a repo to reproduce the error https://github.com/abhay187/lyra-data-persistance-bug.

**Comment by mateonunez** [comment]
> Comment out the plugin import `import { persistToFile } from '@lyrasearch/plugin-data-persistence'` for the page to load. The search result will be printed in the console.

@abhay187 since you're using it on the client side you can't use the server side functionalities such as: `fs`, `path`, etc. To avoid that you can use the **Node Polyfills**. [here a simple package to implement that](https://github.com/sodatea/vite-plugin-node-stdlib-browser)

Furthermore, you should serialize your Lyra instance using the `exportInstance` instead of `persistToFile`, then you can add the logic if you want to download the file via the browser.

Here is a working example of your repository with those arrangements.

![image](https://user-images.githubusercontent.com/11861080/210774775-4862f429-fbb6-49c1-a20c-6188db014de1.png)

**Comment by abhay187** [comment]
Thanks, @mateonunez got it working. Using `importInstance` and `exportInstance`.
Can you share a Cloudflare worker example you have built for reference?

**Comment by paolomainardi** [comment]
Having the same issue while trying to build with esbuild for the browser, even using `importInstance`, you can see it failing here: https://github.com/paolomainardi/hugo-lyra/actions/runs/3976283159/jobs/6816667129 with this commit: https://github.com/paolomainardi/hugo-lyra/pull/28/commits/d1cdb50f49f12a54651ab72c2e7d6bdb299e154d

**Comment by ShogunPanda** [comment]
This has been in fixed in version 0.0.6.
Please note that `importInstance` and `exportInstance` has been removed as they were mostly aliases of `persist` and `restore`.
